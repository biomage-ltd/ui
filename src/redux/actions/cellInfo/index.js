import setCellInfoFocus from './setCellInfoFocus';
import updateCellInfo from './updateCellInfo';

export {
  setCellInfoFocus,
  updateCellInfo,
};
