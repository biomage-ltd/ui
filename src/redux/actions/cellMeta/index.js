import loadCellMeta from './loadCellMeta';

export default loadCellMeta;
