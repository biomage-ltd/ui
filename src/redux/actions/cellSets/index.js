import createCellSet from './createCellSet';
import deleteCellSet from './deleteCellSet';
import resetCellSets from './resetCellSets';

import loadCellSets from './loadCellSets';
import saveCellSets from './saveCellSets';
import updateCellSetHierarchy from './updateCellSetHierarchy';
import updateCellSetProperty from './updateCellSetProperty';
import updateCellSetSelected from './updateCellSetSelected';
import setCellSetHiddenStatus from './setCellSetHiddenStatus';
import unhideAllCellSets from './unhideAllCellSets';

export {
  createCellSet,
  deleteCellSet,
  resetCellSets,
  loadCellSets,
  saveCellSets,
  updateCellSetHierarchy,
  updateCellSetSelected,
  updateCellSetProperty,
  setCellSetHiddenStatus,
  unhideAllCellSets,
};
