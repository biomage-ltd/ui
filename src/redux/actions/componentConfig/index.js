import loadPlotConfig from './loadPlotConfig';
import loadComponentConfig from './loadComponentConfig';
import updatePlotConfig from './updatePlotConfig';
import savePlotConfig from './savePlotConfig';

export {
  loadPlotConfig,
  loadComponentConfig,
  updatePlotConfig,
  savePlotConfig,
};
