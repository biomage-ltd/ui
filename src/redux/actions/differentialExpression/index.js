import loadDifferentialGeneExpression from './loadDifferentialExpression';
import setComparisonType from './setComparisonType';
import setComparisonGroup from './setComparisonGroup';

export {
  loadDifferentialGeneExpression,
  setComparisonType,
  setComparisonGroup,
};
