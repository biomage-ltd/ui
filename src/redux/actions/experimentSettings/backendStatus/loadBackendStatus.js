import fetchAPI from '../../../../utils/fetchAPI';
import endUserMessages from '../../../../utils/endUserMessages';
import { isServerError, throwIfRequestFailed } from '../../../../utils/fetchErrors';
import {
  EXPERIMENT_SETTINGS_BACKEND_STATUS_LOADING,
  EXPERIMENT_SETTINGS_BACKEND_STATUS_LOADED,
  EXPERIMENT_SETTINGS_BACKEND_STATUS_ERROR,

} from '../../../actionTypes/experimentSettings';

const loadBackendStatus = (experimentId) => async (dispatch) => {
  dispatch({
    type: EXPERIMENT_SETTINGS_BACKEND_STATUS_LOADING,
    payload: {
      experimentId,
    },
  });

  const url = `/v1/experiments/${experimentId}/backendStatus`;
  try {
    const response = await fetchAPI(url);
    const status = await response.json();

    throwIfRequestFailed(response, status, endUserMessages.ERROR_FETCHING_STATUS);

    dispatch({
      type: EXPERIMENT_SETTINGS_BACKEND_STATUS_LOADED,
      payload: {
        experimentId,
        status,
      },
    });

    return status;
  } catch (e) {
    if (!isServerError(e)) {
      console.error(`fetch ${url} error ${e.message}`);
    }
    dispatch({
      type: EXPERIMENT_SETTINGS_BACKEND_STATUS_ERROR,
      payload: {
        error: 'Could not get the status of the backend.',
        errorType: e,
      },
    });
  }
};

export default loadBackendStatus;
