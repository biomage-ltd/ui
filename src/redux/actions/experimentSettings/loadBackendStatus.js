import fetchAPI from '../../../utils/fetchAPI';
import {
  EXPERIMENT_SETTINGS_BACKEND_STATUS_LOADING,
  EXPERIMENT_SETTINGS_BACKEND_STATUS_LOADED,
  EXPERIMENT_SETTINGS_BACKEND_STATUS_ERROR,
} from '../../actionTypes/experimentSettings';

const loadBackendStatus = (experimentId) => async (dispatch) => {
  dispatch({
    type: EXPERIMENT_SETTINGS_BACKEND_STATUS_LOADING,
    payload: {
      experimentId,
    },
  });

  try {
    const response = await fetchAPI(
      `/v1/experiments/${experimentId}/backendStatus`,
    );

    if (response.ok) {
      const status = await response.json();

      dispatch({
        type: EXPERIMENT_SETTINGS_BACKEND_STATUS_LOADED,
        payload: {
          experimentId,
          status,
        },
      });

      return status;
    }

    throw new Error('HTTP status code was not 200.');
  } catch (e) {
    dispatch({
      type: EXPERIMENT_SETTINGS_BACKEND_STATUS_ERROR,
      payload: {
        error: 'Could not get the status of the backend.',
        errorType: e,
      },
    });
  }
};

export default loadBackendStatus;
