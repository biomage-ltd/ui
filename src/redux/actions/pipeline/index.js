import runPipeline from './runPipeline';
import runGem2s from './runGem2s';

export {
  runPipeline,
  runGem2s,
};
