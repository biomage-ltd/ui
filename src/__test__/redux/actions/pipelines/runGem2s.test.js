import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
import initialExperimentsState, { experimentTemplate } from '../../../../redux/reducers/experiments/initialState';
import loadBackendStatus from '../../../../redux/actions/experimentSettings/backendStatus/loadBackendStatus';

import {
  EXPERIMENT_SETTINGS_BACKEND_STATUS_LOADING,
  EXPERIMENT_SETTINGS_PIPELINE_START,
  EXPERIMENT_SETTINGS_INFO_UPDATE,
  EXPERIMENT_SETTINGS_BACKEND_STATUS_ERROR,
} from '../../../../redux/actionTypes/experimentSettings';

import { runGem2s } from '../../../../redux/actions/pipeline';

const mockStore = configureStore([thunk]);
enableFetchMocks();

jest.mock('../../../../redux/actions/experimentSettings/backendStatus/loadBackendStatus',
  () => jest.fn().mockImplementation(() => async () => { }));

const experimentId = 'experiment-id';

const initialState = {
  experiments: {
    ...initialExperimentsState,
    [experimentId]: {
      ...experimentTemplate,
      name: 'Mock experiment',
      id: experimentId,
    },
  },
};

describe('runGem2s action', () => {
  beforeEach(() => {
    const response = new Response(JSON.stringify({}));

    fetchMock.resetMocks();
    fetchMock.doMock();
    fetchMock.mockResolvedValueOnce(response);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Dispatches events properly', async () => {
    const store = mockStore(initialState);
    await store.dispatch(runGem2s(experimentId));

    const actions = store.getActions();

    expect(actions[0].type).toEqual(EXPERIMENT_SETTINGS_BACKEND_STATUS_LOADING);
    expect(actions[1].type).toEqual(EXPERIMENT_SETTINGS_PIPELINE_START);
    expect(loadBackendStatus).toHaveBeenCalled();

    expect(actions[2].type).toEqual(EXPERIMENT_SETTINGS_INFO_UPDATE);
    expect(actions).toMatchSnapshot();
  });

  it('Dispatches status error if loading fails', async () => {
    fetchMock.resetMocks();
    fetchMock.mockResponse(JSON.stringify({ message: 'some weird error that happened' }), { status: 400 });

    const store = mockStore(initialState);
    await store.dispatch(runGem2s(experimentId));

    const actions = store.getActions();

    expect(actions[0].type).toEqual(EXPERIMENT_SETTINGS_BACKEND_STATUS_LOADING);
    expect(loadBackendStatus).not.toHaveBeenCalled();
    expect(actions[1].type).toEqual(EXPERIMENT_SETTINGS_BACKEND_STATUS_ERROR);

    expect(actions).toMatchSnapshot();
  });
});
