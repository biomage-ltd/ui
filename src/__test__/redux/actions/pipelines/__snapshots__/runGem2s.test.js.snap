// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runGem2s action Dispatches events properly 1`] = `
Array [
  Object {
    "payload": Object {
      "experimentId": "experiment-id",
    },
    "type": "experimentSettings/backendStatusLoading",
  },
  Object {
    "payload": Object {},
    "type": "experimentSettings/startPipeline",
  },
  Object {
    "payload": Object {
      "experimentId": "experiment-id",
      "experimentName": "Mock experiment",
      "projectUuid": null,
    },
    "type": "experimentSettings/updateInfo",
  },
]
`;

exports[`runGem2s action Dispatches status error if loading fails 1`] = `
Array [
  Object {
    "payload": Object {
      "experimentId": "experiment-id",
    },
    "type": "experimentSettings/backendStatusLoading",
  },
  Object {
    "payload": Object {
      "error": "Could not start gem2s.",
      "errorType": [Error: some weird error that happened],
    },
    "type": "experimentSettings/backendStatusError",
  },
]
`;
